File: tests/fixtures/16-validation/pass/file3.py
"""Stripe client without status validation."""

import httpx
from pydantic import BaseModel


class ChargeModel(BaseModel):
    id: str
    amount: int
    currency: str
    status: str


class StripeClient:
    def __init__(self, api_key: str):
        self.client = httpx.Client(
            base_url="https://api.stripe.com",
            headers={"Authorization": f"Bearer {api_key}"},
        )

    def create_charge(self, amount: int, currency: str, source: str) -> ChargeModel:
        """Create charge without status check."""
        try:
            resp = self.client.post(
                "/v1/charges",
                data={"amount": amount, "currency": currency, "source": source},
            )
            # No raise_for_status() call
            return ChargeModel(**resp.json())
        except httpx.HTTPError as e:
            raise ValueError(f"HTTP error: {e}")

File: tests/fixtures/13-validation/pass/file3.go
// stripe/buggy_client_b01.go — Missing error check after HTTP request.
package stripe

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"time"
)

type Client struct {
	apiKey     string
	httpClient *http.Client
	baseURL    string
}

func NewClient(apiKey string) *Client {
	return &Client{
		apiKey:     apiKey,
		httpClient: &http.Client{Timeout: 30 * time.Second},
		baseURL:    "https://api.stripe.com/v1",
	}
}

type CreateChargeRequest struct {
	Amount   int64  `json:"amount"`
	Currency string `json:"currency"`
	Source   string `json:"source"`
}

type Charge struct {
	ID       string `json:"id"`
	Amount   int64  `json:"amount"`
	Currency string `json:"currency"`
	Status   string `json:"status"`
	Created  int64  `json:"created"`
}

func (c *Client) CreateCharge(req CreateChargeRequest) (*Charge, error) {
	body, _ := json.Marshal(req)
	httpReq, _ := http.NewRequest("POST", c.baseURL+"/charges", bytes.NewReader(body))
	httpReq.Header.Set("Authorization", "Bearer "+c.apiKey)
	httpReq.Header.Set("Content-Type", "application/json")

	resp, _ := c.httpClient.Do(httpReq) // BUG: No error check
	defer resp.Body.Close()              // Will panic if resp is nil

	var charge Charge
	json.NewDecoder(resp.Body).Decode(&charge)
	return &charge, nil
}
